pkg/bootstrap/docker/dockermachine/helper.go

```diff
@@ -147,10 +150,32 @@ func Client(name string) (*docker.Client, error) {
 		client, err = docker.NewVersionedClient(dockerHost, "")
 	}
 	if err != nil {
-		return nil, errors.NewError("could not get Docker client for machine %s", name).WithCause(err)
+		return nil, nil, errors.NewError("could not get Docker client for machine %s", name).WithCause(err)
 	}
 	client.SkipServerVersionCheck = true
-	return client, nil
+
+	var httpClient *http.Client
+	if len(certPath) > 0 {
+		tlscOptions := tlsconfig.Options{
+			CAFile:             filepath.Join(certPath, "ca.pem"),
+			CertFile:           filepath.Join(certPath, "cert.pem"),
+			KeyFile:            filepath.Join(certPath, "key.pem"),
+			InsecureSkipVerify: tlsVerify,
+		}
+		tlsc, tlsErr := tlsconfig.Client(tlscOptions)
+		if tlsErr != nil {
+			return nil, nil, errors.NewError("could not create TLS config client for machine %s", name).WithCause(tlsErr)
+		}
+		httpClient = &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: tlsc,
+			},
+		}
+	}
+
+	engineAPIClient, err := dockerclient.NewClient(dockerHost, "", httpClient, nil)
```

Comment:
I've submitted a PR with 'cluster up' related review changes to @mfojtik 's branch
https://github.com/mfojtik/origin/pull/14